R/utils-shared.R

```diff
@@ -568,6 +570,41 @@ RescaleYTop <- function(object) {
   return(object)
 }
 
+#' Add x-axis labels to five year GW summary plots, and DV hydrographs
+#' having time intervals of one year or more.
+#' @param object A gsplot, plot object.
+#' @param month_label month_label_location
+#' @param date_seq_yr
+#' @return The passed-in gsplot object, with x-axis labeled.
+XAxisLabels <- function(object, month_label, month_label_location, date_seq_yr) {
+  return(
+    mtext(
+      object,
+      text = month_label, at = month_label_location,
+      cex = 0.5, side = 1
+    ) %>%
+      mtext(
+        text = year(date_seq_yr), at = date_seq_yr + (60 * 60 * 24 * 30 * 6),
```

Comment:
this always places the year label around June, but that doesn't mean the year label will show up. For instance, the `dvhydro-example.json` example has 2014-11-20 to 2015-11-20, but you can only see the 2015 label since they are placed in June. Need a more dynamic way to place the labels - like the midpoint of the available data for each year.

E.g.
```r
# do this inside of XAxisLabels; you'd need to pass in plotDates
yr.at <- sapply(year(date_seq_yr), function(y, plotDates){
  which.yr.dates <- which(year(plotDates) == y)
  med.date.each.year <- median(plotDates[which.yr.dates])
  return(med.date.each.year)
}, plotDates = plotDates)
```